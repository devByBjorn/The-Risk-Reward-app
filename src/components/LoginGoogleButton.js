import React from 'react'
import GoogleButtonStyled from '../components_style/LoginGoogleButtonStyled'

const LoginGoogleButton = ({ onClick }) => {

  return (
    <GoogleButtonStyled
      onClick={onClick}
    >Google</GoogleButtonStyled>
  )
}

export default LoginGoogleButton