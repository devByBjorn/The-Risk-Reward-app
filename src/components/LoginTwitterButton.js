import React from 'react'
import TwitterIcon from '@material-ui/icons/Twitter'
import TwitterButtonStyled from '../components_style/LoginTwitterButtonStyled'

const LoginTwitterButton = ({ onClick }) => {

  return (
    <TwitterButtonStyled
      onClick={onClick}
    >
      <TwitterIcon
      />
    </TwitterButtonStyled>
  )
}

export default LoginTwitterButton