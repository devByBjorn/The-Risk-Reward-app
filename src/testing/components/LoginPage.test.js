import React from 'react'
import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import LoginPageDefault, { LoginPage } from '../../components/LoginPage'
import LoginTwitterButton from '../../components/LoginTwitterButton'
import LoginGoogleButton from '../../components/LoginGoogleButton'

const mockStore = configureMockStore([thunk])

test('should render LoginPage with components', () => {
  const wrapper = shallow(<LoginPage />)
  expect(wrapper).toMatchSnapshot()
})

test('should render LoginPage', () => {
  const store = mockStore({})
  const wrapper = shallow(<LoginPageDefault store={store} />)
  expect(wrapper).toMatchSnapshot()
})


// test('Should run googleLogin on button click', () => {
//   const store = mockStore({})
//   const googleLoginSpy = jest.fn()
//   const wrapper = shallow(<LoginPage store={store} googleLogin={googleLoginSpy} />).dive()
//   wrapper.find(LoginGoogleButton).simulate('click')
//   expect(googleLoginSpy).toHaveBeenCalled()
// })