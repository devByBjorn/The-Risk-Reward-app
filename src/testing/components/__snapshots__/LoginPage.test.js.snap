// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage 1`] = `
<LoginPage
  googleLogin={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  twitterLogin={[Function]}
/>
`;

exports[`should render LoginPage with components 1`] = `
<PageMainContainer
  height="auto"
  margin="20rem 10rem"
  mobileMargin="40% 1.5rem"
  overflowY="hidden"
>
  <TopLeftCornerWrap__TopLeftCorner>
    <Logo />
  </TopLeftCornerWrap__TopLeftCorner>
  <FlexContainer
    justifyContent="center"
  >
    <ContentWrapperStyled__ContentWrapper
      alignItems="center"
      flexDirection="column"
    >
      <Paragraph>
        No registration. Just log in. Simple.
      </Paragraph>
      <LoginTwitterButton />
      <LoginGoogleButton />
    </ContentWrapperStyled__ContentWrapper>
  </FlexContainer>
</PageMainContainer>
`;
